sdk/tables/azure-data-tables/tests/test_table_batch.py

```diff
@@ -194,154 +214,208 @@ def test_batch_insert(self, resource_group, location, storage_account, storage_a
-            etag = self.table.create_item(entity, response_hook=lambda e, h: h['etag'])
+            resp = self.table.create_entity(entity=entity)
+            etag = resp['etag']
 
             # Act
             sent_entity = self._create_updated_entity_dict(entity['PartitionKey'], entity['RowKey'])
             batch = self.table.create_batch()
-            batch.update_item(sent_entity, etag=etag, match_condition=MatchConditions.IfNotModified)
-            resp = self.table.commit_batch(batch)
+            batch.update_entity(
+                sent_entity,
+                etag=etag,
+                match_condition=MatchConditions.IfNotModified,
+                mode=UpdateMode.REPLACE
+            )
+            transaction_result = self.table.send_batch(batch)
 
             # Assert
-            self.assertIsNotNone(resp)
-            entity, headers = self.table.read_item(entity['PartitionKey'], entity['RowKey'], response_hook=lambda e, h: (e, h))
+            self._assert_valid_batch_transaction(transaction_result)
+            self.assertIsNotNone(transaction_result.get_entity(sent_entity['RowKey']))
+
+            entity = self.table.get_entity(partition_key=entity['PartitionKey'], row_key=entity['RowKey'])
             self._assert_updated_entity(entity)
-            self.assertEqual(list(resp)[0].headers['Etag'], headers['etag'])
         finally:
             self._tear_down()
 
-    @pytest.mark.skip("pending")
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires Python3")
     @CachedResourceGroupPreparer(name_prefix="tablestest")
     @CachedStorageAccountPreparer(name_prefix="tablestest")
     def test_batch_update_if_doesnt_match(self, resource_group, location, storage_account, storage_account_key):
         # Arrange
         self._set_up(storage_account, storage_account_key)
         try:
             entity = self._create_random_entity_dict()
-            self.table.create_item(entity)
+            self.table.create_entity(entity)
 
             # Act
             sent_entity1 = self._create_updated_entity_dict(entity['PartitionKey'], entity['RowKey'])
 
             batch = self.table.create_batch()
-            batch.update_item(
+            batch.update_entity(
                 sent_entity1,
                 etag=u'W/"datetime\'2012-06-15T22%3A51%3A44.9662825Z\'"',
-                match_condition=MatchConditions.IfNotModified)
-            try:
-                self.table.commit_batch(batch)
-            except PartialBatchErrorException as error:
-                pass  # TODO
-                #self.assertEqual(error.code, 'UpdateConditionNotSatisfied')
-                #self.assertTrue('The update condition specified in the request was not satisfied.' in str(error))
-            else:
-                self.fail('AzureBatchOperationError was expected')
+                match_condition=MatchConditions.IfNotModified
+            )
+
+            with self.assertRaises(HttpResponseError):
```

Comment:
I'm kinda surprised that this is an `HttpResponseError`.
What happens if the batch contained 10 operations and only on of them has a MatchCondition failure - this gives us a failed HTTP status code rather than a `BatchErrorException`?
Seems like this is consistent with the "duplicate row key" error.... wonder how we should handle this. Leave it as is for now, and maybe start a conversation in the channel to see what/if other languages are doing here.

Reply:
Yes it does. I can open that conversation in the channel.